torchdynamo/utils.py

```diff
@@ -51,6 +56,25 @@ def time_wrapper(*args, **kwargs):
         compilation_metrics[key].append(time.time() - t0)
         return r
 
+    def profile_wrapper(*args, **kwargs):
+        global timer_counter
+        datafn = (
+            func.__name__ + f"{next(timer_counter)}.profile"
+        )  # Name the data file sensibly
+        prof = cProfile.Profile()
+        prof.enable()
+        retval = prof.runcall(func, *args, **kwargs)
+        prof.disable()
+        print(f"### Cprofile for {func.__name__} iter {next(timer_counter)} ###")
+        ps = pstats.Stats(prof)
+        ps.sort_stats(pstats.SortKey.TIME).print_stats(20)
+        ps.sort_stats(pstats.SortKey.CUMULATIVE).print_stats(20)
+        prof.dump_stats(datafn)
+        return retval
+
+    if config.use_cprofiler:
+        return profile_wrapper
```

Comment:
anecdotally, did you find it was more useful to get profiles dumped for all these 'typical' hotspot functions and then sift through them?  or easier to look at the timing first and then profile just the one that appears slowest from there

Reply:
The second one for sure. Otherwise there is too much data to sift through and make any sense.